src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala

```diff
@@ -0,0 +1,114 @@
+        case AttributeType.BoolType                       => "-"
+        case AttributeType.DateType                       => "-"
+        case AttributeType.DateTimeType                   => "-"
+        case AttributeType.TimeType                       => "-"
+        case AttributeType.BlobType(size)                 => "-"
+        case AttributeType.StringType(max, min)           => "-"
+        case AttributeType.IntType(max, min, precision)   => "-"
+        case AttributeType.FloatType(max, min, precision) => "-"
+      },
+    )}`"
+
+  private def generateRequestStruct(
+    root: ServiceRoot,
+    operation: CRUD,
+    fields: Iterable[(String, String, String)],
+  ): String =
+    mkCode.lines(
+      s"// ${operation.toString.toLowerCase}${root.name.capitalize}Request contains the client-provided information required to ${operation.toString.toLowerCase} a single ${root.name}",
+      genStructWithAnnotations(s"${operation.toString.toLowerCase}${root.name.capitalize}Request", fields),
+    )
+
+  private[service] def generateRequestStructs(
+    root: ServiceRoot,
+    operations: Set[CRUD],
+    clientAttributes: ListMap[String, Attribute],
+  ): String = {
+    val fields = clientAttributes.map {
+      case (name, attr) =>
+        (name.capitalize, s"*${generateGoType(attr.attributeType)}", generateValidatorAnnotation(attr.attributeType))
+    }
+    mkCode.doubleLines(
+      when(operations.contains(CRUD.Create)) {
+        generateRequestStruct(root, CRUD.Create, fields)
+      },
+      when(operations.contains(CRUD.Update)) {
+        generateRequestStruct(root, CRUD.Update, fields)
+      },
+    )
+  }
+
+  private def generateGoResponseType(attributeType: AttributeType): String =
+    attributeType match {
+      case DateType | TimeType | DateTimeType | _: BlobType => "string"
+      case _                                                => generateGoType(attributeType)
+    }
+
+  private def generateListResponseStructs(root: ServiceRoot, fields: ListMap[String, String]): String =
+    mkCode.lines(
+      s"// list${root.name.capitalize}Element contains a single ${root.name} list element",
+      genStruct(s"list${root.name.capitalize}Element", fields),
+      "",
+      s"// list${root.name.capitalize}Response contains a single ${root.name} list to be returned to the client",
+      genStruct(
+        s"list${root.name.capitalize}Response",
+        ListMap(s"${root.name.capitalize}List" -> s"[]list${root.name.capitalize}Element"),
+      ),
+    )
+
+  private def generateResponseStruct(root: ServiceRoot, operation: CRUD, fields: ListMap[String, String]): String =
+    mkCode.lines(
+      mkCode(
+        s"// ${operation.toString.toLowerCase}${root.name.capitalize}Response contains a",
+        operation match {
+          case Create => s"newly created ${root.name}"
+          case Read   => s"single ${root.name}"
+          case Update => s"newly updated ${root.name}"
```

Comment:
What happens for a delete? I know we know there's never a request body, but feels like you should still deal with the case